vyper/ast.py

```diff
@@ -0,0 +1,278 @@
+from itertools import (
+    chain,
+)
+from typing import (
+    Any,
+    List as ListTyping,
+)
+
+from vyper.exceptions import (
+    CompilerPanic,
+)
+
+
+class VyperNode:
+    __slots__ = ('node_id', 'source_code', 'col_offset', 'lineno')
+    ignored_fields = ['ctx', ]
+    only_empty_fields: ListTyping[Any] = []
+
+    @classmethod
+    def get_slots(cls):
+        return chain.from_iterable(
```

Comment:
You probably want to de-duplicate this with a cast to `set` (also this returns a lazy object that doesn't fully evaluate unless consumed which may not be what you intended.)